src/impl/Status.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "Status.h"
+
+#include <cstring>
+
+namespace milvus {
+
+constexpr int CODE_WIDTH = sizeof(StatusCode);
+
+Status::Status(StatusCode code, const std::string& msg) {
+    // 4 bytes store code
+    // 4 bytes store message length
+    // the left bytes store message string
+    const uint32_t length = (uint32_t)msg.size();
+    auto result = new char[length + sizeof(length) + CODE_WIDTH];
```

Comment:
IMO, we could use 2 private members to hold the `code` and `msg`, then the compiler will automatically generate the default copy/move constructors.
Maybe we could also do the improvement with other PRs.

Reply:
This file is copied from the milvus project. I agree use two members to make it simple. Let's change it by another PR.

Reply:
/lgtm